pkg/v11/resource/chartoperator/current.go

```diff
@@ -8,12 +8,29 @@ import (
 	"github.com/giantswarm/microerror"
 	"github.com/giantswarm/operatorkit/controller/context/resourcecanceledcontext"
 	"github.com/giantswarm/tenantcluster"
+
+	"github.com/giantswarm/cluster-operator/pkg/v11/key"
 )
 
 // GetCurrentState gets the state of the chart in the guest cluster.
 func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interface{}, error) {
 	var err error
 
+	objectMeta, err := r.toClusterObjectMetaFunc(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	// Tenant chart-operator is not deleted so cancel the resource. The operator
+	// will be deleted when the tenant cluster resources are deleted.
```

Comment:
How are the resources deleted though? It is rather that the cluster nodes get ripped apart and all workloads and data vanishes. Is it like that? The comment suggests to me that there is something cleaning properly up, which I think is not the case. These details would also be good to preserve in our internal architecture docs.

Reply:
The way I see it cluster-operator is responsible for deleting the resources in the control plane cluster or we'll have leftovers. aws-operator or the other 2 provider operators are responsible for deleting the tenant clusters so there are no leftovers. 

So we need to delete the cluster-operator cert. But at that point we can't connect to the tenant cluster any more. But this is fine because aws-operator or whichever takes care of deleting the tenant cluster.

For me this is a consequence of the architecture problems with cluster-operator. It should only interact with the control plane cluster. When we fully migrate to app-operator that will be the case. It just creates app CRs in the control plane.

I'll make sure this is covered in the architecture docs I'm writing.